tests/LogNonGenuineHostTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace MongoDB\Tests;
+
+use MongoDB\Client;
+use MongoDB\Driver\Exception\InvalidArgumentException;
+use Psr\Log\AbstractLogger;
+use Psr\Log\LoggerInterface;
+
+use function func_get_args;
+use function MongoDB\addLogger;
+use function MongoDB\removeLogger;
+
+/** @see https://jira.mongodb.org/browse/DRIVERS-2583 */
+class LogNonGenuineHostTest extends TestCase
```

Comment:
This is the only relevant class in the PR. Everything else is from https://github.com/mongodb/mongo-php-library/pull/1158.